File: examples/typescript/operations/parallel/named-branches.ts
import {
  BatchResult,
  DurableContext,
  withDurableExecution,
} from "@aws/durable-execution-sdk-js";

// Plain function branch
async function taskA(ctx: DurableContext): Promise<string> {
  return ctx.step("run-a", async () => "a done");
}

export const handler = withDurableExecution(
  async (event: any, context: DurableContext): Promise<string[]> => {
    const result: BatchResult<string> = await context.parallel("process", [
      // ParallelFunc: plain async function
      taskA,
      // NamedParallelBranch: object with name and func
      { name: "task-b", func: async (ctx) => ctx.step("run-b", async () => "b done") },
    ]);

    return result.getResults();
  },
);

File: examples/python/operations/parallel/named-branches.py
from aws_durable_execution_sdk_python import (
    BatchResult,
    DurableContext,
    ParallelBranch,
    durable_execution,
)


def task_a(ctx: DurableContext) -> str:
    return ctx.step(lambda _: "a done", name="run-a")


def task_b(ctx: DurableContext) -> str:
    return ctx.step(lambda _: "b done", name="run-b")


@durable_execution
def handler(event: dict, context: DurableContext) -> list[str]:
    result: BatchResult[str] = context.parallel(
        [
            # A plain callable runs as an unnamed branch
            task_a,
            # Wrap a callable in ParallelBranch to give the branch a name
            ParallelBranch(func=task_b, name="task-b"),
        ],
        name="process",
    )
    return result.get_results()
